pkg/securitycontext/provider.go

```diff
@@ -65,7 +65,9 @@ func (p SimpleSecurityContextProvider) ModifyHostConfig(pod *v1.Pod, container *
 			for _, group := range pod.Spec.SecurityContext.SupplementalGroups {
 				hostConfig.GroupAdd = append(hostConfig.GroupAdd, strconv.Itoa(int(group)))
 			}
-			if pod.Spec.SecurityContext.FSGroup != nil {
+			// only add the FSGroup to the config if we are not in a remap environment.  For remap environments
+			// is is assumed that the root user is remapped to the FSGroup on the host.
+			if pod.Spec.SecurityContext.FSGroup != nil && !isRemap {
```

Comment:
yes, docker would have to be configured to accept the value.  Using defaults it will fail. It is possible to figure out if it will fail if we read the /etc/subgid file but I was trying to avoid something like that since it also means we need to read (or be told) the daemon argument.